be/src/vec/exprs/lambda_function/varray_match_function.cpp

```diff
@@ -0,0 +1,188 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <glog/logging.h>
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "common/status.h"
+#include "vec/aggregate_functions/aggregate_function.h"
+#include "vec/columns/column.h"
+#include "vec/columns/column_array.h"
+#include "vec/columns/column_nullable.h"
+#include "vec/columns/column_vector.h"
+#include "vec/columns/columns_number.h"
+#include "vec/common/assert_cast.h"
+#include "vec/core/block.h"
+#include "vec/core/column_numbers.h"
+#include "vec/core/column_with_type_and_name.h"
+#include "vec/data_types/data_type.h"
+#include "vec/exprs/lambda_function/lambda_function.h"
+#include "vec/exprs/lambda_function/lambda_function_factory.h"
+#include "vec/exprs/vexpr.h"
+#include "vec/utils/util.hpp"
+
+namespace doris {
+namespace vectorized {
+class VExprContext;
+} // namespace vectorized
+} // namespace doris
+
+namespace doris::vectorized {
+
+///* bool array_match_all/any(array, array<boolean>) *///
+template <bool MATCH_ALL>
+class ArrayMatchFunction : public LambdaFunction {
+    ENABLE_FACTORY_CREATOR(ArrayMatchFunction);
+
+public:
+    ~ArrayMatchFunction() override = default;
+
+    static constexpr auto name = MATCH_ALL ? "array_match_all" : "array_match_any";
+
+    static LambdaFunctionPtr create() { return std::make_shared<ArrayMatchFunction>(); }
+
+    std::string get_name() const override { return name; }
+
+    doris::Status execute(VExprContext* context, doris::vectorized::Block* block,
```

Comment:
The code for processing the first and second columns is quite similar. Consider encapsulating this part of the code into a function and then calling it separately for the first and second columns. 
This would help to avoid this readability warning.